astropy/io/fits/tests/test_fitsdiff.py

```diff
@@ -225,3 +225,17 @@ def test_quiet(self, capsys):
         out, err = capsys.readouterr()
         assert out == ""
         assert err == ""
+
+    def test_path(self, tmpdir):
+        a = np.arange(100).reshape(10, 10)
+        b = np.arange(10000).reshape(100, 100)
+        hdu_a = PrimaryHDU(data=a)
+        hdu_b = PrimaryHDU(data=b)
+        tmp_a = self.temp('testa.fits')
+        tmp_b = self.temp(tmpdir.mkdir('sub').join('testa.fits'))
+        tmp_c = self.temp('sub/')
+        hdu_a.writeto(tmp_a)
+        hdu_b.writeto(tmp_b)
+        numdiff1 = fitsdiff.main(["-q", tmp_a, tmp_b])
+        numdiff2 = fitsdiff.main(["-q", tmp_a, tmp_c])
+        assert numdiff1 == numdiff2
```

Comment:
We need more extensive tests, to test the different cases. You could copy several from `self.data_dir`, modify some files, and test both with globbing, directories, etc.

Reply:
OK..I will do it!

Reply:
while i was writing test cases : `hdu.writeto('sub/testa.fits')` ,since sub/ is absent it should create the path and then write, but it is showing error. isn't this a  bug?

Reply:
I don't *think* `io.fits` creates a directory for you.

Reply:
yes! but logically it should.!

Reply:
Whether it should or not would be a different PR. I don't think it is a bug, so technically it would be a new feature.

Reply:
I don't think it should create the directory. If the user specifies a wrong directory for example, you want to raise an error instead of creating a new tree somewhere in an expected location. So specifying a valid path is the responsibility of the user.